### ISSUE TITLE
MumbleLink written by two gw2 processes

### ISSUE THREAD

**Issue by coderedart** [coderedart/jokolay#25]
when two gw2 instances are started, by default, both of them write to the same MumbleLink shared memory.

when we see a uitick being less than previous tick, at the moment, we assume that previous instance has crashed and we are looking at the data of a new gw2 instance. this will lead to jokolay switching between both windows and acting "crazy" for newbie multi-boxers.

# Case 1: only supporting one active mumblelink per active running gw2 instance
we must explicitly check that the previous gw2 instance is still alive or dead, and if its alive, show a notification or messagebox to inform user that we are following a new gw2 instance now and that they should use different mumblelink names to support jokolay's multiboxing requirements. 
this will be a good way to annoy the newbie multi-boxers to actually use different mumble link names.and it will only be annoying to them and the rest of users would never see this message.

# Case 2: multiple gw2 instances can write to the same mumblelink memory 
we will differentiate between them based on the pid/xid from the data.this is bad. now our mumble data per instance is not updated as often and we are going to "skip" frames. the alternative is to update mumble more often which is pointless when there's only one gw2 instance. 

We will go with Case 1 for now.

**Comment by coderedart** [comment]
we have now shifted to Case 2. 

1. MumbleLinks are just a series of "sources" from which data is read actively.
2. each mumble source keeps track of the previous "ui_tick" and "gw2_pid" (or "gw2_xid" on linux) so that if any of those two variables change, we know that the link has been updated since the last frame. 
3. when a new mumble update is received this frame, we update the live gw2 instances list. these instances, every few seconds, check if the process is alive and get window position and size. 
4. we keep all the live jokolay "game data" separated by each Instance (logged in character, live map etc..).

this way, even if multiple gw2 instances are updating the same mumble link, we will be fine due to unique ID (xid or pid). although, this will slow down the updates because we will only read the mumblelink memory once per frame and thus only one instance data is read and the other might be missed due to data races, so multiple links are still preferred.
